src/SenseNet.Security.EFCSecurityStore/SecurityStorage.cs

```diff
@@ -120,9 +120,13 @@ internal void _cleanupMembership()
         }
 
 
-        private const string SelectUnprocessedActivityIds = @"SELECT Id AS Value FROM [EFMessages] WHERE ExecutionState != 'Done'
+        private const string SelectUnprocessedActivityIds = @"DECLARE @lastInserted INT
+DECLARE @ident INT
+SELECT @lastInserted = MAX(Id) FROM EFMessages
+SELECT @ident = CONVERT(int, IDENT_CURRENT('EFMessages'))
+SELECT Id, Id AS [Value] FROM [EFMessages] WHERE ExecutionState != 'Done'
```

Comment:
Why the double select? `SELECT Id, Id`
And also the `SELECT 0` below.

Reply:
It seems terrible for the first time but it is not a bug but a special expectation. The EFC uses EfcIntSet (our own) that expects id-value pairs.